pypsa/statistics/abstract.py

```diff
@@ -245,14 +245,12 @@ def _filter_active_assets(self, n: Network, c: str, obj: Any) -> Any:
         idx = self._get_component_index(obj, c)
         if not self.is_multi_indexed:
             mask = n.get_active_assets(c)
-            idx = mask.index[mask].intersection(idx)
-            return obj.loc[idx]
+            return obj.loc[mask.index[mask].intersection(idx)]
```

Comment:
mmh, what's the difference now? I am missing something? It's the same as before wo defining `idx`

Reply:
no, the difference is that here we are NOT updating `idx`, but using the full index in each investment period.

Reply:
oh yes, true, sorry, what a stupid error!